### PR TITLE
Jetpack Backup: update daily backup card once a backup is completed

### PR DESCRIPTION
**Pull request by Initsogar**
Resolves https://github.com/Automattic/jetpack-backup-team/issues/159
Inspired by #89638

If a user starts a new backup on demand, or if they open the backup page when a backup is in progress, when it completes we show a `Backup just completed` view, telling them the backup will be ready in a few minutes. If the user wants to view the files or restore to this backup, they will need to refresh the page.

The idea of this PR is to update the backup status as soon as possible, instead of let them be in the `Backup just completed` state forever. This approach involves to refetch the backups from the Activity Log every 5 seconds once the backup is completed, so it will automatically update when a new backup arrived.

## Proposed Changes

* Pass `refetch` from `useRewindableActivityLogQuery` upstream to allow manual refresh.
* Refetch backups from Activity Log every 5 seconds after a backup is completed until we got the new backup.
* Updated the component import and all references from `MostRecentStatus` to `DailyBackupStatus` for improved readability and maintainability.

## Demo

https://github.com/Automattic/wp-calypso/assets/1488641/9a29e021-5f8d-4ddf-addb-a8bffa84c0d2

## Testing Instructions

* Spin up a Calypso or Jetpack Cloud live branch
* Select a site with Jetpack VaultPress Backup
* Click on `Back up now`
* Wait until it finished. It should display a `Backup just completed` view, and after some time it will automatically update into a `Latest backup` view.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by mavegaf** [comment]
I tested it, and it works perfectly for the first backup.

If a backup just finished and without refreshing the site I run "Backup up now", after it finishes again, instead of showing the screen  "Backup Just completed",  it shows the screen "Latest backup" but with the previous backup, and after some time appears the new backup

**Comment by Initsogar** [comment]
> I tested it, and it works perfectly for the first backup.
> 
> If a backup just finished and without refreshing the site I run "Backup up now", after it finishes again, instead of showing the screen "Backup Just completed", it shows the screen "Latest backup" but with the previous backup, and after some time appears the new backup

Nice catch! Thank you! I will take a look tomorrow and see how to fix it.

File: client/components/jetpack/daily-backup-status/index.jsx [hunk 1]
```diff
@@ -86,6 +98,31 @@ const DailyBackupStatus = ( {
...
+
+	// Effect for handling backup updates and Activity Log fetching intervals.
+	useEffect( () => {
+		// Set lastBackup on initial load or clear interval if backup's rewindId changes.
+		if ( backup && ! lastBackup ) {
+			setLastBackup( backup );
+		} else if ( backup && lastBackup && backup.rewindId !== lastBackup.rewindId ) {
+			backupPreviouslyInProgress.current = null;
+			clearActivityLogInterval();
+			setLastBackup( backup );
```
Comment (Initsogar) [inline client/components/jetpack/daily-backup-status/index.jsx:112]: In this scenario, we already have the new backup, so we can set it as the last backup to fix https://github.com/Automattic/wp-calypso/pull/89751#issuecomment-2071084187.

**Comment by Initsogar** [comment]
@mavegaf it should be working now. Could you take another look?

**Review by mavegaf — APPROVED** [review]

**Pull request merged**